networkx/algorithms/community/tests/test_modularity_max.py

```diff
@@ -287,17 +287,36 @@ def test_resolution_parameter_impact():
     assert naive_greedy_modularity_communities(G, resolution=gamma) == expected
 
 
-def test_n_communities_parameter():
+def test_cutoff_parameter():
     G = nx.circular_ladder_graph(4)
 
     # No aggregation:
     expected = [{k} for k in range(8)]
-    assert greedy_modularity_communities(G, n_communities=8) == expected
+    assert greedy_modularity_communities(G, cutoff=8) == expected
 
     # Aggregation to half order (number of nodes)
     expected = [{k, k + 1} for k in range(0, 8, 2)]
-    assert greedy_modularity_communities(G, n_communities=4) == expected
+    assert greedy_modularity_communities(G, cutoff=4) == expected
 
     # Default aggregation case (here, 2 communities emerge)
     expected = [frozenset(range(0, 4)), frozenset(range(4, 8))]
-    assert greedy_modularity_communities(G, n_communities=1) == expected
+    assert greedy_modularity_communities(G, cutoff=1) == expected
+
+
+def test_best_n():
+    G = nx.barbell_graph(5, 3)
+
+    # Same result as without enforcing n_communities:
+    best_n = 3
+    expected = [frozenset(range(5)), frozenset(range(8, 13)), frozenset(range(5, 8))]
+    assert greedy_modularity_communities(G, best_n=best_n) == expected
+
+    # One additional merging step:
+    best_n = 2
+    expected = [frozenset(range(8)), frozenset(range(8, 13))]
+    assert greedy_modularity_communities(G, best_n=best_n) == expected
+
+    # Tne additional merging steps:
```

Comment:
```suggestion
    # Two additional merging steps:
```

I think? (based on the previous comment)